README.md

```diff
@@ -17,21 +17,71 @@ It requires an LSM enabled orchestrator, with no ssl or authentication enabled,
 
 ### First case: using a remote orchestrator
 
-This plugin is built around the remote_orchestrator fixture.
+This plugin is built around the remote_orchestrator fixture and the `ServiceInstance` class.
 
-A typical testcase using this plugin looks as follows:
+Typical test cases using this fixture will look as follow:
+1. `test_full_cycle` relies on the helper function `service_full_cycle` to create, update and delete a service instance, following its progress asynchronously.  Multiple services are created by invoking the helper function multiple times.  All the coroutines created are passed to `util.execute_scenarios` which will take care of deploying all the services, and report the failures if any.  This asynchronous test case allows us to test multiple services at the same time, without a complete reset of the orchestrator.
+2. `test_transient_state` simply synchronously deploys and delete a service, and illustrates the usage of the `wait_for_state` method.
 ```python
+import uuid
+import inmanta_lsm.model
+from pytest_inmanta import plugin
+
+from pytest_inmanta_lsm import lsm_project, remote_orchestrator, service_instance, util
+
+SERVICE_NAME = "vlan-assignment"
+
+
+async def service_full_cycle(
+    remote_orchestrator: remote_orchestrator.RemoteOrchestrator,
+    router_ip: str,
+    interface_name: str,
+    address: str,
+    vlan_id: int,
+    vlan_id_update: int,
+) -> None:
+    # Create an async service instance object
+    instance = service_instance.ServiceInstance(
+        remote_orchestrator=remote_orchestrator,
+        service_entity_name=SERVICE_NAME,
+    )
+
+    # Create the service instance on the remote orchestrator
+    await instance.create(
+        {
+            "router_ip": router_ip,
+            "interface_name": interface_name,
+            "address": address,
+            "vlan_id": vlan_id,
+        },
+        wait_for_state="up",
+        timeout=60,
+    )
+
+    # Update the vlan id
+    await instance.update(
+        [
+            inmanta_lsm.model.PatchCallEdit(
+                edit_id=str(uuid.uuid4()),
+                operation=inmanta_lsm.model.EditOperation.replace,
+                target="vlan_id",
+                value=vlan_id_update,
+            ),
+        ],
+        wait_for_state="up",
+        timeout=60,
+    )
+
+    # Delete the instance
+    await instance.delete(wait_for_state="terminated", timeout=60)
 
-def test_full_cycle(project, remote_orchestrator):
+
+def test_full_cycle(project: plugin.Project, remote_orchestrator: remote_orchestrator.RemoteOrchestrator) -> None:
```

Comment:
I would not explain this more sophisticated packing mechanism as the base case. 

I would start with a simple testcase first (either async or not) and then go into the more subtle packing case .

Reply:
I reorganized it a bit, is it better now?